### PR TITLE
infra: removed hash info and added server logs step to ci

### PR DESCRIPTION
**Pull request by SmailBarkouch**

### CODE REVIEW

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -14,6 +14,7 @@ jobs:
     env:
       GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
       LOCKBOOK_CI: 1
```
Comment (Parth) [inline .github/workflows/integration.yml:16]: [github](https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables) has an env var for CI already we should just use that and avoid thinking and dealing with this.

File: utils/dev-tool/src/main.rs [hunk 2]
```diff
@@ -52,40 +55,26 @@ enum Commands {
...
-    let hash_info_dir = utils::hash_infos_dir(&dev_dir);
+    let server_db_dir = utils::server_dbs_dir(&dev_dir);
 
-    fs::create_dir_all(&dev_dir).unwrap();
-    fs::create_dir_all(&hash_info_dir).unwrap();
     fs::create_dir_all(&target_dir).unwrap();
-    fs::create_dir_all(&server_dbs_dir).unwrap();
+    fs::create_dir_all(&server_db_dir).unwrap();
 
     env::set_var("CARGO_TARGET_DIR", &target_dir.to_str().unwrap());
```
Comment (Parth) [inline utils/dev-tool/src/main.rs:75]: I think you can just set this in `integration.yml` once and then not have this program think about this, ever, right?

File: utils/dev-tool/src/utils.rs [hunk 3]
```diff
@@ -1,10 +1,9 @@
-use serde::{Deserialize, Serialize};
 use std::env::{self, VarError};
-use std::fs::{self, File};
-use std::io::Write;
 use std::path::{Path, PathBuf};
 use std::process::{Command, Output, Stdio};
 
+pub const SERVER_PORT: u16 = 8501;
```
Comment (Parth) [inline utils/dev-tool/src/utils.rs:5]: Why are you overriding this at all? Why not use the default `test.env` values?

File: utils/dev-tool/src/server.rs [hunk 4]
```diff
@@ -1,80 +1,54 @@
...
-            .env("INDEX_DB_LOCATION", &server_db_dir)
-            .current_dir(&tool_env.root_dir)
-            .stderr(Stdio::null())
-            .stdout(Stdio::null())
-            .spawn()
-            .unwrap();
+    let mut run_result = Command::new(tool_env.target_dir.join("release/lockbook-server"))
+        .env("SERVER_PORT", SERVER_PORT.to_string())
+        .env("INDEX_DB_LOCATION", &tool_env.server_db_dir)
+        .env("LOG_PATH", &tool_env.server_db_dir)
```
Comment (Parth) [inline utils/dev-tool/src/server.rs:18]: Why are you overriding this at all? Why not use the default `test.env` values?

File: utils/dev-tool/src/server.rs [hunk 5]
```diff
@@ -1,80 +1,54 @@
...
 
-        thread::sleep(Duration::from_millis(5000));
+    thread::sleep(Duration::from_millis(5000));
 
-        if run_result.try_wait().unwrap().is_none() {
-            break;
-        }
+    if run_result.try_wait().unwrap().is_some() {
+        panic!("Server failed to start on port: '{}'", SERVER_PORT)
     }
```
Comment (Parth) [inline utils/dev-tool/src/server.rs:29]: I thought about this more, and I'm pretty certain we don't want this tool to build then run. I think we want to cargo run, and check if the server is up by polling `http://localhost:8000/get-build-info`.

Either the server will be up, then you're good to go, or the command will terminate early (fail to compile or start).

File: clients/egui/build.rs [hunk 6]
```diff
@@ -1,6 +1,8 @@
 #[cfg(windows)]
 fn main() -> std::io::Result<()> {
-    winres::WindowsResource::new().set_icon("lockbook.ico").compile()
+    winres::WindowsResource::new()
+        .set_icon("lockbook.ico")
+        .compile()
```
Comment (Parth) [inline clients/egui/build.rs:5]: If you rebase this will disappear from your diff

**Review by Parth — CHANGES_REQUESTED** [review]

File: utils/dev-tool/src/server.rs (hunk 4 above)
Comment (SmailBarkouch) [reply utils/dev-tool/src/server.rs:18]: Fair point, defaults can be used.

**Review by Parth — APPROVED** [review]

**Pull request merged**